Integration with sensu-plugins-go-library

sayakju opened the pull request:
Update `sensu-pagerduty-handler` to use new `sensu-plugins-go-library`

Commit:
Integration with sensu-plugins-go-library

nikkictl commented on Gopkg.toml:
```diff
@@ -30,12 +30,8 @@
   revision = "fe8f9c4593d05f029afeb2c08c3e220120306506"
 
 [[constraint]]
-  name = "github.com/sensu/sensu-go"
```
https://github.com/sensu/sensu-slack-handler/pull/19#discussion_r289935497

sayakju replied:
Correct me if I am wrong. I think github.com/sensu/sensu-plugins-go-library library will pull in github.com/sensu/sensu-go as it has dependency on it.

nikkictl replied:
Yes, sensu-plugins-go-library will pull in sensu-go as a dependency. However this repository, sensu-pagerduty-handler is _directly_ dependent on the sensu-go api (specifically it imports corev2.Event). If there were a change to that api in sensu-go that is required for this handler, the dependency would have to be bumped in sensu-plugins-go-library first.

While this does compile, sensu-go should be treated as a direct dependency, rather than a transitive one (see https://github.com/golang/dep/blob/master/docs/FAQ.md#what-is-a-direct-or-transitive-dependency).

sayakju replied:
Makes sense. Added the direct dependency in the latest commit.

Commit:
1) Add direct dependency on github.com/sensu/sensu-go
2) Removed redundant dependency on github.com/stretchr/testify
3) Added default value of parameter "token"

fguimond commented:
@nikkiki @calebhailey - @sayakju and I don't have permissions to merge the changes. Can you please merge them when you have a chance? Or please let us know what the process should be.
